File: src/main/scala/assignment3/e0/WorldSnapshot.java
package assignment3.e0;

import java.util.ArrayList;

/**
 * 
 *
 */
public class WorldSnapshot {
	
	private P2d[] posList;
	private double time;
	
	public WorldSnapshot(ArrayList<P2d> pos, double time) {
		posList = new P2d[pos.size()];
		int index = 0;
		for (P2d p: pos) {
			posList[index++] = p;
		}
		this.time = time;
	}
	
	public P2d[] getPosList() {
		return posList;
	}
	
	public double getTime() {
		return time;
	}
}

File: src/main/scala/assignment3/e1/WorldSnapshot.scala
package assignment3.e1

import scala.collection.mutable.ArrayBuffer

class WorldSnapshot(val pos: ArrayBuffer[P2d], var time: Double) {
  var posList: ArrayBuffer[P2d] = new ArrayBuffer(pos.length)
  var index = 0

  for (p <- pos) {
    posList += p
  }

  def getPosList: ArrayBuffer[P2d] = posList

  def getTime: Double = time
}